.circleci/config.yml

```diff
@@ -32,100 +24,117 @@ jobs:
             make -j$(nproc) -C ./builddir all
-      - run: #Only run make check on go1.11 as it introduces a new formatting rule we're following
-          name: Run checks
-          command: make -C ./builddir check
+      - run:
+          name: check code
+          command: |-
+            go get -u golang.org/x/lint/golint
+            make -j$(nproc) -C ./builddir check
+
   go1.11-alpine:
     <<: *defaults
     docker:
-      - image: golang:1.11-alpine
+      - image: sylabsio/golang:1.11-alpine
     steps:
+      - attach_workspace:
+          at: /go
       - run:
-          name: Install git
+          name: Build Singularity
           command: |-
-            apk add --update git
+            ./mconfig -p /usr/local
+            make -j$(nproc) -C ./builddir all
+            go get -u golang.org/x/lint/golint
+            make -j$(nproc) -C ./builddir check
       - run:
-          name: checkout
+          name: check code
           command: |-
-            git clone https://github.com/sylabs/singularity.git .
-      - restore_cache:
-          key: dep-cache-{{ checksum "Gopkg.lock" }}
-          paths:
-            - vendor
+            go get -u golang.org/x/lint/golint
+            make -j$(nproc) -C ./builddir check
+
+  singularity_test:
+    machine: true
+    steps:
+      - attach_workspace:
+          at: ~/go
       - run:
-          name: Install dep
+          name: Setup environment
           command: |-
-            go get -u github.com/golang/dep/cmd/dep
-      - run:
-          name: Install golint
+            echo 'export  GOPATH=$HOME/go' >> $BASH_ENV
+            echo 'export  GOROOT=/usr/local/go' >> $BASH_ENV
+            echo 'export  GOBIN=$HOME/go/bin' >> $BASH_ENV
+            echo 'export  GOCACHE=off' >> $BASH_ENV
+            echo 'export PATH=$GOPATH/bin:$GOBIN:$GOROOT/bin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/usr/local/sbin' >> $BASH_ENV
+      - run:
+          name: Update Go to 1.11
+          working_directory: /tmp
           command: |-
-            go get -u golang.org/x/lint/golint
+            wget https://dl.google.com/go/go1.11.linux-amd64.tar.gz
+            sudo rm -rf /usr/local/go
+            sudo tar -C /usr/local -xzf go1.11.linux-amd64.tar.gz
+            sudo chown -R circleci:circleci ~/go
       - run:
-          name: Set Up Vendor Directory
+          name: fetch deps
+          working_directory: /tmp
           command: |-
-            if [ ! -d vendor ]; then
-              dep ensure -vendor-only
-            fi
-      - save_cache:
-          key: dep-cache-{{ checksum "Gopkg.lock" }}
-          paths:
-            - vendor
-      - run:
-          name: Install C dependencies
-          command: apk add --update alpine-sdk automake libtool linux-headers linux-vanilla libarchive-dev util-linux-dev libuuid openssl-dev gawk
+            mkdir -p ${GOBIN}
+            curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
```

Comment:
why do we install dep even though its already fetched from the `get_source` pipeline?
wouldnt that mean we already have our `/vendor` directory populated before this and thus no longer need to install dep via `curl` ?

Reply:
From Slack:
> Andre: yes but why do we need to install it on the machine? is it needed for running go build?
i thought its not needed as long as `/vendor` has the files already?

> Eduardo Arango [3:09 AM]
`./mconfig` will fail if it is not in path
is just to make `./mconfig` happy
when we merge Cedric PR on the vendor folder
I can then modify that part

Reply:
this will addressed once #2200 is merged